src/main/kotlin/org/wycliffeassociates/otter/jvm/app/ui/takemanagement/viewmodel/RecordableViewModel.kt

```diff
@@ -0,0 +1,106 @@
+package org.wycliffeassociates.otter.jvm.app.ui.takemanagement.viewmodel
+
+import com.github.thomasnield.rxkotlinfx.observeOnFx
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.subjects.PublishSubject
+import javafx.beans.property.SimpleBooleanProperty
+import javafx.beans.property.SimpleObjectProperty
+import javafx.collections.FXCollections
+import javafx.collections.ObservableList
+import org.wycliffeassociates.otter.common.data.workbook.AssociatedAudio
+import org.wycliffeassociates.otter.common.data.workbook.Take
+import org.wycliffeassociates.otter.common.domain.content.EditTake
+import org.wycliffeassociates.otter.common.domain.content.RecordTake
+import org.wycliffeassociates.otter.common.domain.content.Recordable
+import org.wycliffeassociates.otter.jvm.app.ui.takemanagement.TakeContext
+import tornadofx.FX.Companion.messages
+import tornadofx.*
+
+open class RecordableViewModel(private val audioPluginViewModel: AudioPluginViewModel) {
+
+    val recordableProperty = SimpleObjectProperty<Recordable?>()
+    var recordable by recordableProperty
+
+    private val disposables = CompositeDisposable()
+
+    val selectedTakeProperty = SimpleObjectProperty<Take?>()
+    val contextProperty = SimpleObjectProperty<TakeContext>(TakeContext.RECORD)
+    val showPluginActiveProperty = SimpleBooleanProperty(false)
+    var showPluginActive by showPluginActiveProperty
+
+    val snackBarObservable: PublishSubject<String> = PublishSubject.create()
+
+    val alternateTakes: ObservableList<Take> = FXCollections.observableList(mutableListOf())
+
+    init {
+        recordableProperty.onChange {
+            // TODO
+        }
+
+        selectedTakeProperty.onChange {
+            // TODO
+        }
+    }
+
+    fun recordNewTake() {
+        recordable?.let {
+            contextProperty.set(TakeContext.RECORD)
+            showPluginActive = true
+            audioPluginViewModel
+                .record(it)
+                .observeOnFx()
+                .doOnSuccess { result ->
+                    showPluginActive = false
+                    when (result) {
+                        RecordTake.Result.NO_RECORDER -> snackBarObservable.onNext(messages["noRecorder"])
+                        else -> {}
```

Comment:
Use an exhaustive `when` here instead of `else`; it'll only add one line.

Reply:
done